antiplag/tests/tests_email.py

```diff
@@ -0,0 +1,47 @@
+from django.test import TestCase
+from django.conf import settings
+from django.test import Client
+
+# Create your tests here.
+
+
+class EmailTestCase(TestCase):
+    def test_invalid_email(self):
+        """Verify that an invalid email returns correct error message"""
+        c = Client()
+        response = c.post(
+            "/api/submissions/",
+            {"email": "invalid", "text": "This is my text"},
+            content_type="application/json",
+        )
+        self.assertEqual(
+            response.content.decode("utf-8"),
+            '{"error":"Unrecognized email address format."}',
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def test_valid_email(self):
+        """Verify that a valid email passes through validation"""
+        c = Client()
+
+        response = c.post(
```

Comment:
As far  as I remember - during retrospective we were talking whether we should or should not test third-party applications. Please discuss with @OliverKanat how to mock the `send_email` function.

Reply:
Email validation is being tested in views properly for all possible cases. Consulted with Oliver.